### DISCUSSION TITLE
Has anyone ever needed to increment/decrement via api call a column?

### DISCUSSION THREAD

**Discussion by macropay-solutions** [laravel/framework#52803] [Ideas]
Hello,

We saw some threads recently about increments and thought to ask this.

**Comment by ibrahimBougaoua** [comment]
**Hi, 
Yes, You might need this for use cases such as updating the stock of a product, adjusting a user's score,update qty in e commerce website....etc.**

**Reply by macropay-solutions** [reply]
We are analyzing if we should include this into our free crud lib on update in the form of 
`{"incrementable_column": "++5", "updateable_colum":"6"}`
or
`{"incrementable_column": "--5", "updateable_colum":"6"}`

in the body, but we have an issue with the  incrementOrDecrement function from model. It does not sync changes from the extra param as mentioned here https://github.com/laravel/framework/discussions/52786 , meaning the observers will not see updateable_colum as changed.

Overwriting the function is NOK because it changes between L8 and L11.

So the dilemma is: should we implement something that we know will not work as expected 100%?

**Reply by macropay-solutions** [reply]
[We ended up implementing this only in the paid version](https://github.com/macropay-solutions/laravel-lumen-crud-wizard/blob/main/README.md#ii3-updateupsertincrement-resource) by custom code.

Demo: https://github.com/macropay-solutions/laravel-crud-wizard-demo/blob/fe6dc32ac9a6f01f8202e42ed9b60efbf448c505/app/Http/Controllers/ResourceController.php#L36
